asset/js/setup.js

```diff
@@ -10,30 +10,26 @@ function setup() {
 }
 
 function setupWithSearch(siteData) {
-  const routeArray = jQuery.map(siteData.pages, object => object.src);
-  const titleArray = jQuery.map(siteData.pages, object => object.title);
-  const { typeahead } = VueStrap.components;
+  const { frontmattersearch } = VueStrap.components;
   const vm = new Vue({
     el: '#app',
     components: {
-      typeahead,
+      frontmattersearch,
     },
     data() {
       return {
-        searchData: titleArray,
+        searchData: siteData.pages,
       };
     },
     methods: {
       searchCallback(match) {
```

Comment:
`searchData` is used [here](https://github.com/MarkBind/vue-strap/blob/411dcba73bebb57d2bb1b2ead7cfd6708631f72f/src/Typeahead.vue#L103).

'searchCallback` is used [here](https://github.com/MarkBind/vue-strap/blob/411dcba73bebb57d2bb1b2ead7cfd6708631f72f/src/Typeahead.vue#L149).

The `Typeahead` component as it is now won't support our front matter search because it is only doing matches on a primitive array. The PR https://github.com/MarkBind/vue-strap/pull/17 adds a method that allow matching on an array of objects.

Reply:
The 2 lines that you linked are:
- `return this.data.filter(value => {`
- `this.onHit(this.items[this.current], this)`

Am I missing something?

> The `Typeahead` component as it is now won't support our front matter search because it is only doing matches on a primitive array. The PR MarkBind/vue-strap#17 extends the `Typeahead` component by adding a method that allow matching on an array of objects.

The "array of objects" is from what we write to **siteData.json** right? Can't we preprocess the `title` and `keywords`? It's more maintainable than a custom "subclass"-ed component.

Reply:
Okay I think it's clearer here. It's used in the HTML to display the searchbar: https://github.com/MarkBind/vue-strap/pull/17

That's correct. Hmm.. How could we preprocess that in a way that is understandable by `Typeahead.vue`? Currently it does matches from simple arrays like follows:
`['Oregon', 'Arizona', 'New York']`

An example of how `Typeahead.vue` can be used now is shown [here](http://yuche.github.io/vue-strap/#typeahead)

What we need is a way for it to match against titles and keywords from the pages that we have.

Reply:
> Okay I think it's clearer here. It's used in the HTML to display the searchbar: MarkBind/vue-strap#17

Ah, I was searching in the code. Thanks for pointing that out, much clearer now.

> How could we preprocess that in a way that is understandable by `Typeahead.vue`?

We could concatenate the `title` and `keywords`.
Can you post a screenshot of how the proposed `FrontMatterSearch` component looks?

Reply:
Ah, got objects to work!

```html
<typeahead
  :data="searchData"
  :template="titleTemplate"
  template-name="title"
  placeholder="Search"
  match-case
></typeahead>
```

```js
data () {
  return {
    searchData: [
      {
        title: 'Title 1',
        keywords: ['Keyword', 'drowyeK'],
        value: function () { return [this.title].concat(this.keywords).join(' '); },
        indexOf: function (query) { return this.value().indexOf(query) },
        toLowerCase: function () { return this.value().toLowerCase(); }
      },
      {
        title: 'Title 2',
        keywords: ['Keyword', 'drowyeK'],
        value: function () { return [this.title].concat(this.keywords).join(' '); },
        indexOf: function (query) { return this.value().indexOf(query) },
        toLowerCase: function () { return this.value().toLowerCase(); }
      }
    ],
    'titleTemplate': '{{ item.title }}<br><sub>{{ item.keywords }}</sub>',
  }
},
```

<img width="340" alt="screenshot 2018-04-15 01 12 04" src="https://user-images.githubusercontent.com/14091939/38770726-0a6e3696-404a-11e8-99c9-e8affb9635ba.png">

This works by reverse-engineering **Typeahead.vue**:

```js
primitiveData () {
  if (this.data) {
    return this.data.filter(value => {
      value = this.matchCase ? value : value.toLowerCase()
      var query = this.matchCase ? this.value : this.value.toLowerCase()
      return this.matchStart ? value.indexOf(query) === 0 : value.indexOf(query) !== -1
    }).slice(0, this.limit)
  }
}
```